### DISCUSSION TITLE
Q: State of Debugging

### DISCUSSION THREAD

**Discussion by DanPartelly** [odin-lang/Odin#2325] [Q&A]
Hey,

Just some fast questions:

1. What is the state of debugging on Linux? 
2. What notable debugging tools support Odin (Linux/Windows) gdb / lldb, valgrind, whatever. 
3. What LLVM sanitizers / X-ray tooling are supported by Odin, if any.

With many thanks.

**Comment by Beefster09** [comment]
1. In my experience, not great. Best I've really gotten out of gdb is a stack trace without line numbers. Usually gets all sorts of confused when crossing language boundaries because the symbol table seems to be corrupted or something.
2. gdb kind of supports it
3. no idea

**Reply by z64** [reply]
> In my experience, not great. Best I've really gotten out of gdb is a stack trace without line numbers. Usually gets all sorts of confused when crossing language boundaries because the symbol table seems to be corrupted or something.

In this scenario, are you building your odin code with `-debug`? What you describe sounds like behavior when you compile without debug info or with optimizations enabled (`-o:speed`, etc.)

**Reply by Beefster09** [reply]
I'm one of the users affected by the bug where compiling with debug creates a faulty binary (sometimes), so that's probably a big reason why gdb explodes even with debug symbols on.

**Comment by MarcusE1W** [comment]
I used lldb for my first time and I found it surprisingly helpful. 
You can set break points, step line by line through the code and inspect variables. 
There is probably more, but I have not figured that out yet.

**Reply by felipebueno** [reply]
> I used lldb for my first time and I found it surprisingly helpful.

Yeah, me too! And lldb also offers a gui to make the debugging experience more enjoyable (if you prefer visual debuggers like me :)

![image](https://github.com/odin-lang/Odin/assets/241271/3de22cb4-98f8-47b7-bce8-70e218e021f9)

**Reply by pehagg** [reply]
How do you invoke lldb and what build command are you using? I've tried with `odin build . -debug -o:none` and launch it with `lldb ./myprog`, but the sources just aren't there.
